symbolic-base/src/ZkFold/Symbolic/Data/UInt.hs

```diff
@@ -389,7 +389,7 @@ instance
                 (zs, b) <- flip runStateT b0 $ traverse StateT (Haskell.zipWith (fullSub r) is js)
                 d <- newAssigned (\v -> v i' - v j')
                 s'0 <- newAssigned (\v -> v d + v b + fromConstant t)
-                (s', _) <- splitExpansion (highRegisterSize @(BaseField c) @n @r) 1 s'0
+                (s', _) <- splitExpansion (highRegisterSize @(BaseField c) @n @r) (r + 1 -! highRegisterSize @(BaseField c) @n @r) s'0
```

Comment:
Nice catch! 
Though I think it's better to fix the definition of `s'0` to use `2 ^ highRegisterSize - 1` instead of `t` in order to have less constraints